.github/workflows/update-opengrep.yml

```diff
@@ -0,0 +1,67 @@
+name: Update Opengrep
+
+on:
+  schedule:
+    # Run every Monday at 10:00 AM UTC
+    - cron: '0 10 * * 1'
+  workflow_dispatch: # Allow manual triggering
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  update-opengrep:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@08eba0b27e820071cde6df949e0beb9ba4906955 # v4.3.0
+
+      - name: Setup Node.js
+        uses: actions/setup-node@49933ea5288caeca8642d1e84afbd3f7d6820020 # v4.4.0
+        with:
+          node-version: '24.x'
+
+      - name: Run opengrep update script
+        id: update
+        continue-on-error: true
+        run: node third_party/opengrep/update-opengrep.js
+
+      - name: Check for changes and create PR
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          if git diff --quiet; then
```

Comment:
this code should be in roll_opengrep.py. See roll_abseil.py for an example

Reply:
ok, so we have a standard for doing these kind of stuff